recipe/build-arrow.sh

```diff
@@ -12,6 +12,10 @@ EXTRA_CMAKE_ARGS=""
 if [ "$(uname)" == "Linux" ]; then
   SYSTEM_INCLUDES=$(echo | ${CXX} -E -Wp,-v -xc++ - 2>&1 | grep '^ ' | awk '{print "-isystem;" substr($1, 1)}' | tr '\n' ';')
   EXTRA_CMAKE_ARGS=" -DARROW_GANDIVA_PC_CXX_FLAGS=${SYSTEM_INCLUDES}"
+else
+  # See https://conda-forge.org/docs/maintainer/knowledge_base.html#newer-c-features-with-old-sdk
+  EXTRA_CMAKE_ARGS=" -DARROW_CXXFLAGS=\"${CXXFLAGS} -D_LIBCPP_DISABLE_AVAILABILITY\""
```

Comment:
The escaping here is invalidated by the following line. Shouldn't it be sufficient to add `D_LIBCPP_DISABLE_AVAILABILITY` to CXXFLAGS directly?

Reply:
I'm trying to sync with the [work](https://github.com/apache/arrow/commit/9d6598108c4fd93be89f8f5becaa7cb66f929fe7#diff-b81d923c288d441a9b33891d4cbc60323da63f1c906bdff67db1ff944296e834) from @pitrou upstream, but yeah, bash quoting rules got the better of me...